politeiawww/user/mysql/mysql.go

```diff
@@ -1011,9 +1019,137 @@ func (m *mysql) Close() error {
 	return m.userDB.Close()
 }
 
+// EmailHistoriesSave creates or updates the email histories.
+//
+// EmailHistoriesSave satisfies the user MailerDB interface.
+func (m *mysql) EmailHistoriesSave(histories map[uuid.UUID]user.EmailHistory) error {
+	log.Tracef("EmailHistoriesSave: %v", histories)
+
+	if m.isShutdown() {
+		return user.ErrShutdown
+	}
+
+	ctx, cancel := ctxWithTimeout()
+	defer cancel()
+
+	for userID, history := range histories {
+		var (
+			update bool
+			id     string
+		)
+		err := m.userDB.
+			QueryRowContext(ctx,
+				"SELECT userID FROM email_histories WHERE userID = ?", userID).
+			Scan(&id)
+		switch err {
+		case nil:
+			// Email history already exists for this user, update it.
+			update = true
+		case sql.ErrNoRows:
+			// Email history doesn't exist for this user, create new one.
+		default:
+			// All other errors
+			return fmt.Errorf("lookup: %v", err)
+		}
+
+		// Make email history blob
+		ehb, err := user.EncodeEmailHistory(history)
+		if err != nil {
+			return fmt.Errorf("convert email history to DB: %w", err)
+		}
+		eb, err := m.encrypt(user.VersionEmailHistory, ehb)
+		if err != nil {
+			return err
+		}
+
+		// Save email history
+		if update {
+			_, err := m.userDB.ExecContext(ctx,
+				`UPDATE email_histories SET hBlob = ? WHERE userID = ?`,
+				eb, userID)
+			if err != nil {
+				return fmt.Errorf("update: %v", err)
+			}
+		} else {
+			_, err := m.userDB.ExecContext(ctx,
+				`INSERT INTO email_histories (userID, hBlob) VALUES (?, ?)`,
+				userID, eb)
+			if err != nil {
+				return fmt.Errorf("create: %v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// EmailHistoriesGet returns the email histories for the specified users.
```

Comment:
If I read this comment I have no idea what happens if a email history doesn't exist for the user.